src/index.ml

```diff
@@ -474,32 +514,24 @@ struct
     let generation =
       match log with None -> Int63.zero | Some log -> IO.get_generation log.io
     in
-    let log_async_path = Layout.log_async ~root in
-    (* If we are in readonly mode, the log_async will be read during sync_log so
-       there is no need to do it here. *)
-    if (not readonly) && IO.exists log_async_path then (
-      let io =
-        IO.v ~flush_callback ~fresh ~generation:Int63.zero ~fan_size:Int63.zero
-          log_async_path
-      in
-      let entries = Int63.div (IO.offset io) entry_sizeL in
-      Log.debug (fun l ->
-          l "[%s] log_async file detected. Loading %a entries"
-            (Filename.basename root) Int63.pp entries);
-      (* If we are not in fresh mode, we move the contents of log_async to
-         log. *)
-      if not fresh then
-        Option.iter
-          (fun log ->
-            let append_io = IO.append log.io in
-            iter_io
-              (fun e ->
-                Tbl.replace log.mem e.key e.value;
-                Entry.encode e append_io)
-              io;
-            IO.flush log.io;
-            IO.clear ~generation ~reopen:false io)
-          log);
+    (* recover [log_async] entries *)
+    let () =
+      let log_async_path = Layout.log_async ~root in
+      (* - If we are in readonly mode, the log_async will be read
+           during sync_log so there is no need to do it here. *)
+      if (not readonly) && (not fresh) && IO.exists log_async_path then
+        let io =
+          IO.v ~flush_callback ~fresh ~generation ~fan_size:Int63.zero
+            log_async_path
+        in
+        (* in fresh mode, we need to wipe the existing [log_async] file. *)
+        if fresh then IO.clear ~generation ~reopen:false io
```

Comment:
This `fresh = true` branch is unreachable because of the `(not fresh)` test above

Reply:
woops I forgot to remove it - well spotted!